python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -1115,6 +1115,15 @@ class Cast(Expr):
     def __init__(self, dtype: plc.DataType, value: Expr) -> None:
         super().__init__(dtype)
         self.children = (value,)
+        if (
+            self.dtype.id() == plc.TypeId.STRING
+            or value.dtype.id() == plc.TypeId.STRING
+        ):
+            raise NotImplementedError(
+                "Need to implement cast to/from string separately."
+            )
```

Comment:
I think `cast` doesn't support any "compound" types so perhaps:

```
if plc.traits.is_compound(self.dtype) or plc.traits.is_compound(value.dtype):
    raise ...
```

Reply:
We might also consider implementing (like the `is_supported_operation` for compiled binops) a `cudf::is_supported_cast` function (`cast_ops.cu` has an `is_supported_cast<From, To>()` function already)

Reply:
I think this would be the cleanest way forward 👍

Reply:
Now that #15470 is merged do we have what we need here?

Reply:
Yes! this should be generally implemented now with https://github.com/rapidsai/cudf/pull/16232/commits/dc0ebbbd8f2c76a8ab00f20046597aedd788279c